### PR TITLE
AM2R: Fix several tower connecting rooms 

### PR DESCRIPTION
**Pull request by JeffGainsNGames**
multiple missing morph ball requirements, tricks listed as mid-air morph rather than glide, and added a new method for crossing Zeta East Access

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,7 +7,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [8.3.x] - 2024-08-??
 
-- Nothing yet! Add changes here.
+### AM2R
+
+#### Logic Database
+
+##### The Tower
+
+- Added: Ext. Zeta Nest East Access now has an IBJ(intermediate) + Spider method of crossing the room from the right.
```
Comment (Miepee) [inline CHANGELOG.md:16]:
```suggestion
- Added: Ext. Zeta Nest East Access now has an IBJ (intermediate) + Spider method of crossing the room from the right.
```

File: randovania/games/am2r/logic_database/The Tower.txt [hunk 2]
```diff
@@ -404,20 +404,25 @@ Extra - liquid_info: {'liquid_type': 0, 'liquid_level': 0}
...
   > Horizontal Dock to Exterior Zeta Nest East
       Any of the following:
-          Space Jump or Can Use Spider Ball
+          Space Jump
           Damage Boost (Beginner) and Normal Damage ≥ 50
-          Hi-Jump Boots and Morph Glide (Advanced)
+          Hi-Jump Boots and Morph Ball and Morph Glide (Advanced)
           Diagonal Infinite Bomb Jumping (Intermediate) and Can IBJ and Can Use Bombs
+          All of the following:
+              Can Use Spider Ball
```
Comment (Miepee) [inline randovania/games/am2r/logic_database/The Tower.txt:424]: do we assume the player just knows they can crawl "through" the spikes?

File: randovania/games/am2r/logic_database/The Tower.txt [hunk 3]
```diff
@@ -506,7 +511,7 @@ Extra - liquid_info: {'liquid_type': 0, 'liquid_level': 0}
   > Middle
       Any of the following:
           Space Jump
-          Hi-Jump Boots and Morph Glide (Advanced)
+          Hi-Jump Boots and Morph Ball and Morph Glide (Advanced)
```
Comment (Miepee) [inline randovania/games/am2r/logic_database/The Tower.txt:514]: what about spider along the ceiling, fall down, then morph glide?

File: randovania/games/am2r/logic_database/The Tower.txt [hunk 4]
```diff
@@ -569,7 +574,7 @@ Extra - liquid_info: {'liquid_type': 0, 'liquid_level': 0}
               Any of the following:
                   Power Grip
                   Walljump (Intermediate) and Can Walljump
-          Hi-Jump Boots and Mid-Air Morph (Intermediate)
+          Hi-Jump Boots and Morph Ball and Morph Glide (Intermediate)
```
Comment (Miepee) [inline randovania/games/am2r/logic_database/The Tower.txt:577]: what about IBJ? if that alone doesnt suffice, IBJ+Spider?

File: randovania/games/am2r/logic_database/The Tower.txt (hunk 3 above)
Comment (JeffGainsNGames) [reply randovania/games/am2r/logic_database/The Tower.txt:514]: image
as this only would ever get you halfway, I don't see the worth in adding it

Comment (Miepee) [reply randovania/games/am2r/logic_database/The Tower.txt:514]: ~~but what if you start in the middle~~
Fair enough.

**Review by Miepee — APPROVED** [review]

**Pull request merged**